src/main/scala/devices/uart/UART.scala

```diff
@@ -230,23 +230,23 @@ trait HasUARTTopModuleContents extends MultiIOModule with HasUARTParameters with
     UARTCtrlRegs.txfifo -> NonBlockingEnqueue(txq.io.enq),
     UARTCtrlRegs.rxfifo -> NonBlockingDequeue(rxq.io.deq),
 
-    UARTCtrlRegs.txctrl -> Seq(
-      RegField(1, txen),
-      RegField(stopCountBits, nstop)),
-    UARTCtrlRegs.rxctrl -> Seq(RegField(1, rxen)),
-    UARTCtrlRegs.txmark -> Seq(RegField(txCountBits, txwm)),
-    UARTCtrlRegs.rxmark -> Seq(RegField(rxCountBits, rxwm)),
+    UARTCtrlRegs.txctrl -> RegFieldGroup("txctrl",Some("Serial transmit control"),Seq[RegField](
+      RegField(1, txen, RegFieldDesc("txen","Transmit enable", reset=Some(0))),
+      RegField(stopCountBits, nstop, RegFieldDesc("nstop","Number of stop bits", reset=Some(0))))),
+    UARTCtrlRegs.rxctrl -> Seq(RegField(1, rxen, RegFieldDesc("txen","Receive enable", reset=Some(0)))),
+    UARTCtrlRegs.txmark -> Seq(RegField(txCountBits, txwm, RegFieldDesc("txcnt","Transmit watermark level", reset=Some(0)))),
+    UARTCtrlRegs.rxmark -> Seq(RegField(rxCountBits, rxwm, RegFieldDesc("rxcnt","Receive watermark level", reset=Some(0)))),
 
-    UARTCtrlRegs.ie -> Seq(
-      RegField(1, ie.txwm),
-      RegField(1, ie.rxwm)),
+    UARTCtrlRegs.ie -> RegFieldGroup("ie",Some("Serial interrupt enable"),Seq[RegField](
+      RegField(1, ie.txwm, RegFieldDesc("txwm_ie","Transmit watermark interrupt enable", reset=Some(0))),
+      RegField(1, ie.rxwm, RegFieldDesc("rxwm_ie","Receive watermark interrupt enable", reset=Some(0))))),
 
-    UARTCtrlRegs.ip -> Seq(
-      RegField.r(1, ip.txwm),
-      RegField.r(1, ip.rxwm)),
+    UARTCtrlRegs.ip -> RegFieldGroup("ip",Some("Serial interrupt pending"),Seq[RegField](
+      RegField.r(1, ip.txwm, RegFieldDesc("txwm_ip","Transmit watermark interrupt pending")),
```

Comment:
these should be `volatile=true`

Reply:
Also I don't think you need to say `Seq[RegField]`... does `Seq(...)` not work?

Reply:
Seq() does in fact work - I was just copying the incantation from an example...